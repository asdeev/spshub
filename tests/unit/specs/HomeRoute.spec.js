import { createLocalVue, shallowMount } from '@vue/test-utils';
import { expect } from 'chai';
import Vuetify from 'vuetify';
import Vuex from 'vuex';
import initialState from '~/store/state';
import testMap from '../../testMap';
import homeFixture from './fixtures/home';

const localVue = createLocalVue();
localVue.use(Vuex);
localVue.use(Vuetify);

// TODO: Fix failing tests
describe('HomeRoute', () => {
  let state;

  const build = () => {
    const wrapper = shallowMount(testMap.HomeRoute, {
      localVue,
      state: new Vuex.Store({ state })
    });

    return {
      wrapper,
      carousel: () => wrapper.find('.home-route__carousel'),
      imagesArray: () => wrapper.find('.home-route__images')
    };
  };

  beforeEach(() => {
    state = { ...initialState };
  });

  it('renders the component', () => {
    const { wrapper } = build();

    expect(wrapper.html()).to.matchSnapshot();
  });

  it('renders the main child components', () => {
    const { carousel } = build();

    expect(carousel().exists()).to.be.true;
  });

  it('renders the carousel images correctly', () => {
    state.home = homeFixture;

    const { imagesArray } = build();

    const image0 = imagesArray().at(0);
    const image3 = imagesArray().at(3);

    expect(image0.attributes().src).to.be.equal(
      './assets/images/education.jpg'
    );
    expect(image3.attributes().src).to.be.equal(
      './assets/images/baby-camel.jpg'
    );
  });
});
