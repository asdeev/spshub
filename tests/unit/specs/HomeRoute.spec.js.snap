// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeRoute renders the component 1`] = `"<v-carousel-stub activeclass=\\"v-item--active\\" mandatory=\\"true\\" cycle=\\"true\\" delimitericon=\\"$vuetify.icons.delimiter\\" height=\\"500\\" hidecontrols=\\"true\\" interval=\\"6000\\" nexticon=\\"$vuetify.icons.next\\" previcon=\\"$vuetify.icons.prev\\" class=\\"home-route__carousel\\"><v-carousel-item-stub src=\\"./assets/images/education.jpg\\" class=\\"home-route__images\\"></v-carousel-item-stub><v-carousel-item-stub src=\\"./assets/images/elephant-herd.jpg\\" class=\\"home-route__images\\"></v-carousel-item-stub><v-carousel-item-stub src=\\"./assets/images/seattle-night.jpg\\" class=\\"home-route__images\\"></v-carousel-item-stub><v-carousel-item-stub src=\\"./assets/images/baby-camel.jpg\\" class=\\"home-route__images\\"></v-carousel-item-stub><v-carousel-item-stub src=\\"./assets/images/school-children.jpg\\" class=\\"home-route__images\\"></v-carousel-item-stub></v-carousel-stub>"`;
